sbt-scala-native/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePluginInternal.scala

```diff
@@ -322,6 +322,16 @@ object ScalaNativePluginInternal {
       IO.read(nativeNativelib.value / "target")
     },
     nativeMode := "debug",
+    nativeDebug := {
+      val binary = abs(nativeLink.value)
+      val lldb   = abs(nativeLLDB.value)
+      val logger = streams.value.log
+      Process(lldb +: Seq("-f", binary, "-o", "run"))
+        .run(logger, connectInput = true)
+        .exitValue
+
+      ()
+    },
```

Comment:
But you'll have to turn it off for release builds. I'm not sure if extra complexity is worth it here. Especially considering that C/C++ code in nativelib are mostly wrappers.